app/src/main/java/com/google/android/sambadocumentsprovider/provider/SambaDocumentsProvider.java

```diff
@@ -360,7 +365,20 @@ public void onTaskFinished(
         return cursor;
       }
     } catch (AuthFailedException e) {
-      return buildErrorCursor(projection, R.string.view_folder_denied);
+      if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.O && DocumentMetadata.isShareUri(uri)) {
+        Intent authIntent = new Intent();
```

Comment:
Move logic in the "if" block to a separate method.

Create a static method AuthActivity.createIntent() and make AuthActivity.SHARE_URI_KEY private.